hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/ActionRepository.java

```diff
@@ -490,4 +501,12 @@ Long countByRolloutAndRolloutGroupAndStatusNotIn(JpaRollout rollout, JpaRolloutG
     @Query("DELETE FROM JpaAction a WHERE a.id IN ?1")
     void deleteByIdIn(Collection<Long> actionIDs);
 
+    /** Updates the externalRef of an action by its actionId.
+     * @param actionId for which the externalRef is being updated.
+     * @param externalRef value of the external reference for the given action id.
+     */
+    @Modifying
+    @Transactional
+    @Query("UPDATE JpaAction a SET a.externalRef = :externalRef WHERE a.id = :actionId")
+    void updateExternalRef(@Param("actionId") Long actionId, @Param("externalRef") String externalRef);
```

Comment:
I'm not sure if this is the best way to do it. Personally I would move this logic to the service. I would just find the action from the repository, update it in the service and save it.

take a look here to understand more what I mean:
https://stackoverflow.com/questions/32295688/spring-data-jpa-update-method

Reply:
another point is that you don't check that the action exists first, which most probably will cause problems.

Reply:
Reason being this column update approach is that it reduces one more query. And since it execute a sql query, it does not affect if actionId does not exist.

If we look closely at the usecase, the consumer of this api gets to consume this only because a valid action was generated. So checking if action exist will be too paranoid in my opinion.

Reply:
Personally I would prefer to still have the extra query. But ok, that's also a valid point. Let's leave it like this for now, and let's see what the other guys will say in the "big" pull request :)